app/services/the_tvdb_service.rb

```diff
@@ -38,6 +69,29 @@ def mapping_data
                    end
   end
 
+  def missing_thumbnail_data
+    @mtd ||= Mapping.where(external_site: 'thetvdb')
+                    .joins('LEFT OUTER JOIN episodes ON mappings.item_id = episodes.media_id AND mappings.item_type = episodes.media_type')
+                    .where(episodes: { thumbnail_file_name: nil })
+                    .distinct.pluck(:item_type, :item_id, :external_site, :external_id)
+                    .each_with_object({}) do |(item_type, item_id, external_site, external_id), acc|
+                      acc[item_type] ||= {}
+                      acc[item_type][item_id] ||= {}
+                      acc[item_type][item_id][external_site] = external_id
+                    end
+  end
+
+  def currently_airing_data
+    @cad ||= Mapping.where(external_site: %w[thetvdb/series thetvdb/season])
+                    .joins('LEFT OUTER JOIN anime ON mappings.item_id = anime.id').merge(Anime.current)
+                    .pluck(:item_type, :item_id, :external_site, :external_id)
+                    .each_with_object({}) do |(item_type, item_id, external_site, external_id), acc|
```

Comment:
You repeat this logic in multiple places, DRY it out

Reply:
not sure how to DRY this type of thing out tbh.

Reply:
See that big part where you do `pluck` and then `each_with_object`, which is duplicated?  Stick it in a method.